QEfficient/peft/onnx_transforms.py

```diff
@@ -0,0 +1,56 @@
+# -----------------------------------------------------------------------------
+#
+# Copyright (c) 2024 Qualcomm Innovation Center, Inc. All rights reserved.
+# SPDX-License-Identifier: BSD-3-Clause
+#
+# ----------------------------------------------------------------------------
+
+from typing import Tuple
+
+import onnx
+
+from QEfficient.base.onnx_transforms import OnnxTransform
+
+
+class AdapterWeightsToInputsTransform(OnnxTransform):
+    @classmethod
+    def apply(cls, model: onnx.ModelProto, *, adapter_name: str, **kwargs) -> Tuple[onnx.ModelProto, bool]:
+        transformed = False
+        removed_initializers = []
+
+        # Find nodes with lora weights as inputs
+        weight_suffix = f".{adapter_name}.weight"
+        lora_weight_nodes = {
+            inp: node for node in model.graph.node for inp in node.input if inp.endswith(weight_suffix)
+        }
+
+        for i, weight in enumerate(model.graph.initializer):
+            if weight.name.endswith(weight_suffix):
+                transformed = True
+
+                # Create input/output for lora weights
+                new_weight_name = weight.name[: -len(weight_suffix)] + ".weight"
+                type_proto = onnx.helper.make_tensor_type_proto(weight.data_type, shape=list(weight.dims))
+                inp = onnx.ValueInfoProto(name=new_weight_name, type=type_proto)
+                out = onnx.ValueInfoProto(name=new_weight_name + "_RetainedState", type=type_proto)
+                model.graph.input.append(inp)
+                model.graph.output.append(out)
+
+                # Create a node that connects input -> output
+                node = onnx.helper.make_node("Identity", [inp.name], [out.name], new_weight_name + "_identity")
+                model.graph.node.append(node)
+
+                # Rename weight input
+                lora_weight_node = lora_weight_nodes[weight.name]
+                for j, inp in enumerate(lora_weight_node.input):
+                    if inp.endswith(weight_suffix):
+                        lora_weight_node.input[j] = new_weight_name
```

Comment:
have a map of `lora_weight_node_state = {node: True/False}` If True skip the for loop.
I think currently this will be called again in case the node inputs were already updated.

Reply:
This will not be called again, because we're renaming the weight name, so it will not match the `endswith(...)` condition. If all the inputs are already renamed, it will not match any of the inputs and it is already skipping over.